NExtends/Primitives/String.extensions.cs

```diff
@@ -9,555 +9,577 @@
-		/// Annee10 => Annee
-		/// </summary>
-		/// <param name="str"></param>
-		/// <returns></returns>
-		public static string RemoveOperand(this string str)
-		{
-			return new string(str.TakeWhile(c => !Char.IsNumber(c)).ToArray());
-		}
-
-		/// <summary>
-		/// Annee10 => Annee11
-		/// Annee => Annee2
-		/// </summary>
-		/// <param name="str"></param>
-		/// <param name="increment (optional)"></param>
-		/// <returns></returns>
-		public static string IncrementOperand(this string str, int increment = 1)
-		{
-			return String.Format("{0}{1}", str.RemoveOperand(), str.GetOperand() + increment);
-		}
-
-		/// <summary>
-		/// Remove special characters to transform input into an valid file name
-		/// Replaces spaces by dots
-		/// </summary>
-		/// <param name="input"></param>
-		/// <returns></returns>
-		public static string ToFileName(this string input)
-		{
-			return string.IsNullOrEmpty(input) ? input : Regex.Replace(input, @"[^\w. ]+", "").Replace(" ", ".");
-		}
-	}
+    public static class StringExtensions
+    {
+        private const string DEFAULT_ELLIPSIS = "...";
+
+        public static DateTime ParseJsonDate(this string date)
+        {
+            var result = new DateTime();
+            //http://stackoverflow.com/questions/5521553/best-way-to-convert-javascript-date-to-net-date
+            if (!DateTime.TryParseExact(date.Substring(0, 24), "ddd MMM d yyyy HH:mm:ss", CultureInfo.InvariantCulture, DateTimeStyles.AssumeLocal, out result))
+            {
+                //Ie, of course
+                //http://stackoverflow.com/questions/1877788/javascript-date-to-c-sharp-via-ajax pour ie
+                result = DateTime.ParseExact(date, "ddd MMM d HH:mm:ss UTCzzzzz yyyy", CultureInfo.InvariantCulture);
+            }
+            return result;
+        }
+
+        public static string Format(this string s, params object[] args)
+        {
+            return args != null ? string.Format(s, args) : s;
+        }
+
+        //Validation des GUID qui arrivent en paramètre
+        static Regex isGuid = new Regex(@"^(\{){0,1}[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}(\}){0,1}$", RegexOptions.Compiled);
+
+        //Validation des numéro de sécu INSEE
+        //version de gabsoftware
+        //http://www.developpez.net/forums/d677820/php/langage/regex/verification-numero-securite-sociale/
+        private const string REGULAR_INSEE_NUMBER = @"([1-3])[\s\.\-]?(\d{2})[\s\.\-]?(0\d|[2-35-9]\d|[14][0-2])[\s\.\-]?(0[1-9]|[1-9]\d|2[abAB])[\s\.\-]?(\d{3})[\s\.\-]?(\d{3})[\s\.\-\|]{0,3}";
+        private const string TEMPORARY_INSEE_NUMBER = @"(?<temp>[7-9][\s\.\-]?\d{2}[\s\.\-]?\d{2}[\s\.\-]?\d{2}[\s\.\-]?\d{3}[\s\.\-]?\d{3}[\s\.\-\|]?)";
+        private const string KEY_INSEE_NUMBER = @"(?<key>[0-8]\d|9[0-7])?";
+
+        static Regex isINSEENumber = new Regex(String.Format(@"^(({0})|({1})){2}$", REGULAR_INSEE_NUMBER, TEMPORARY_INSEE_NUMBER, KEY_INSEE_NUMBER), RegexOptions.Compiled);
```

Comment:
all theses regex should be in another place I think. It's not really StringExtension methods, but more specific functional tests. We should provide another class for theses methods